posts/inside-rust/2025-01-17-this-development-cycle-in-cargo-1.85.md

```diff
@@ -0,0 +1,388 @@
+
+## Plugin of the cycle
+
+Cargo can't be everything to everyone,
+if for no other reason than the compatibility guarantees it must uphold.
+Plugins play an important part of the Cargo ecosystem and we want to celebrate them.
+
+Our choice for this cycle is [clippy-sarif](https://crates.io/crates/clippy-sarif).  While not a plugin, it wraps `cargo clippy`, converting the lint output into [Static Analysis Results Interchange Format (SARIF)](https://sarifweb.azurewebsites.net) which GitHub can use to post comments on a PR review for lints, making them easier to access than reading a log file.
+
+Thanks to [epage](https://github.com/epage) for the suggestion!
+
+[Please submit your suggestions for the next post.](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Plugin.20of.20the.20Dev.20Cycle/near/420703211)
+
+## Implementation
+
+### Rustflags and caching
+
+*Update from [1.84](https://blog.rust-lang.org/inside-rust/2024/12/13/this-development-cycle-in-cargo-1.84.html#rustflags-and-caching)*
+
+As a recap, Cargo throws out the cache on changes to `RUSTFLAGS`,
+instead of creating separate entries in the cache.
+The problem is `RUSTFLAGS` is opaque to Cargo and different flags have different caching needs.
+The remaining problem is `--remap-path-prefix` which is used to make two different builds look the same.
+The parameters to the flag are unique to each build and the cache key ends up in the final binary,
+preventing `--remap-path-prefix` from serving its purpose.
+
+<!-- 2024-12-03 -->
+We talked about this a Cargo team meeting.
+Once we have built-in support for `--remap-path-prefix` through
+[trim-paths](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#profile-trim-paths-option),
+we could just point people to that instead.
+However, there are open issues with trim-paths and its unclear how long that will take to become available.
+We also could go back to the fix for this from the previous attempt and parse `RUSTFLAGS`
+(see [#6966](https://github.com/rust-lang/cargo/pull/6966))
+but we're uncomfortable with that because parsing of command-lines is context-sensitive.
+We ended up with a middle-ground where we apply a rough heuristic where if something looks like `--remap-path-prefix`,
+we do not include `RUSTFLAGS` in our cache key.
+
+This was merged in [#14830](https://github.com/rust-lang/cargo/pull/14830)
+
+### Report progress in the taskbar
+
+Previously, [Gordon01](https://github.com/Gordon01)
+picked up work on [ConEmu's](https://conemu.github.io/en/AnsiEscapeCodes.html#ConEmu_specific_OSC) OSC 9;4 which is an ANSI escape code they defined to report progress through the Windows taskbar icon
+([#14615](https://github.com/rust-lang/cargo/pull/14615)).
+Support for reporting this has extended to applications like systemd
+([systemd#34929](https://github.com/systemd/systemd/pull/34929))
+while support for reporting this has extended to applications like
```

Comment:
This sentence is a duplicate of the previous sentence and should probably be replaced with just "and"

Reply:
I had the wrong word in there, it wasn't mean to say "application" but "terminal".